### PR TITLE
protobuf examples

### PR DESCRIPTION
**Pull request by famartinrh**

### CODE REVIEW

File: protobuf-bean/src/main/java/io/apicurio/registry/examples/simple/protobuf/ProtobufBeanExample.java [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+        props.putIfAbsent(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG, "true");
+        props.putIfAbsent(ConsumerConfig.AUTO_COMMIT_INTERVAL_MS_CONFIG, "1000");
+        props.putIfAbsent(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
+        props.putIfAbsent(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class.getName());
+        // Use the Apicurio Registry provided Kafka Deserializer for Protobuf
+        props.putIfAbsent(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, ProtobufKafkaDeserializer.class.getName());
+
+        // Configure Service Registry location
+        props.putIfAbsent(SerdeConfig.REGISTRY_URL, REGISTRY_URL);
+        props.putIfAbsent(SerdeConfig.EXPLICIT_ARTIFACT_GROUP_ID, "default");
```
Comment (EricWittmann) [inline protobuf-bean/src/main/java/io/apicurio/registry/examples/simple/protobuf/ProtobufBeanExample.java:187]: Is this needed?  I assumed the deserializer wouldn't need this.

File: protobuf-find-latest/src/main/java/io/apicurio/registry/examples/simple/protobuf/ProtobufFindLatestExample.java [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+        RegistryClient client = RegistryClientFactory.create(REGISTRY_URL);
+        System.out.println("Manually creating the artifact in Apicurio Registry");
+        //because the default ArtifactResolverStrategy is TopicIdStrategy the artifactId is in the form of topicName-value
+        InputStream protofile = Thread.currentThread().getContextClassLoader().getResourceAsStream("person.proto");
+        client.createArtifact("default", topicName + "-value", protofile);
+        System.out.println();
+
+        // Create the producer.
+        Producer<Object, AddressBook> producer = createKafkaProducer();
+        // Produce 5 messages.
```
Comment (EricWittmann) [inline protobuf-find-latest/src/main/java/io/apicurio/registry/examples/simple/protobuf/ProtobufFindLatestExample.java:85]: 2 messages :)

File: protobuf-find-latest/src/main/java/io/apicurio/registry/examples/simple/protobuf/ProtobufFindLatestExample.java [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+        props.putIfAbsent(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG, "true");
+        props.putIfAbsent(ConsumerConfig.AUTO_COMMIT_INTERVAL_MS_CONFIG, "1000");
+        props.putIfAbsent(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
+        props.putIfAbsent(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class.getName());
+        // Use the Apicurio Registry provided Kafka Deserializer for Protobuf
+        props.putIfAbsent(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, ProtobufKafkaDeserializer.class.getName());
+
+        // Configure Service Registry location
+        props.putIfAbsent(SerdeConfig.REGISTRY_URL, REGISTRY_URL);
+        props.putIfAbsent(SerdeConfig.EXPLICIT_ARTIFACT_GROUP_ID, "default");
```
Comment (EricWittmann) [inline protobuf-find-latest/src/main/java/io/apicurio/registry/examples/simple/protobuf/ProtobufFindLatestExample.java:190]: Needed?

File: protobuf-find-latest/src/main/java/io/apicurio/registry/examples/simple/protobuf/ProtobufFindLatestExample.java [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+        props.putIfAbsent(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, StringSerializer.class.getName());
+        // Use the Apicurio Registry provided Kafka Serializer for Protobuf
+        props.putIfAbsent(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, ProtobufKafkaSerializer.class.getName());
+
+        // Configure Service Registry location
+        props.putIfAbsent(SerdeConfig.REGISTRY_URL, REGISTRY_URL);
+        props.putIfAbsent(SerdeConfig.EXPLICIT_ARTIFACT_GROUP_ID, "default");
+
+        // Find and use the latest artifact in the registry for the corresponding GroupId and ArtifactId
+        props.putIfAbsent(SerdeConfig.FIND_LATEST_ARTIFACT, Boolean.TRUE);
```
Comment (EricWittmann) [inline protobuf-find-latest/src/main/java/io/apicurio/registry/examples/simple/protobuf/ProtobufFindLatestExample.java:165]: You aren't setting AUTO_REGISTER here.  The default for that is `true` I think?  How do these two configs work together? E.g. what happens if I set `AUTO_REGISTRY=true` **and** `FIND_LATEST_ARTIFACT=true`?

**Comment by EricWittmann** [comment]
Cool. :)  Just had a few questions inline.

File: protobuf-bean/src/main/java/io/apicurio/registry/examples/simple/protobuf/ProtobufBeanExample.java (hunk 1 above)
Comment (famartinrh) [reply protobuf-bean/src/main/java/io/apicurio/registry/examples/simple/protobuf/ProtobufBeanExample.java:187]: no, it's not need, I'll remove it

File: protobuf-find-latest/src/main/java/io/apicurio/registry/examples/simple/protobuf/ProtobufFindLatestExample.java (hunk 4 above)
Comment (famartinrh) [reply protobuf-find-latest/src/main/java/io/apicurio/registry/examples/simple/protobuf/ProtobufFindLatestExample.java:165]: the default for `auto-register` is false, and the same for `find-latest`. How this configs work together is defined in the `DefaultSchemaResolver` implementation. For now it does: 
- 1. if `auto-register` is true try to create the artifact
- 2. if `find-latest` is true try to find the artifact by the resolved `ArtifactReference`
- 3. if the schema is available beforehand (avro and protobuf) try to find the artifact by content
- 4 fallback (jsonschema would be the only one to reach here) to find the artifact by `ArtifactReference`

I went with this approach to be slightly similar to the old serdes, where the find by content was the default. But we can change this whatever other way we think it makes more sense.

**Comment by famartinrh** [comment]
I just force pushed, but no worries I fixed that typo

**Pull request merged**